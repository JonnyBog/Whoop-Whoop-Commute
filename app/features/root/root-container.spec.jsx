import React from 'react';
import rendererShallow from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';

import RootContainer from 'features/root/root-container';

describe('Features', () => {
  describe('Root Container', () => {
    it('should export a connected component', () => {
      expect(RootContainer).toBeDefined();
    });

    const mockStore = configureMockStore();
    const store = mockStore({});
    let props;

    beforeEach(() => {
      props = {};
    });

    afterEach(() => {
      props = null;
    });

    it('should connect', () => {
      const wrapper = shallow(
        <MemoryRouter>
          <RootContainer {...props} store={store} />
        </MemoryRouter>
      );

      expect(wrapper.instance().props).toEqual(expect.any(Object));
    });

    it('should render correctly', () => {
      const shallowRenderer = rendererShallow.createRenderer();
      const tree = shallowRenderer.render((
        <Provider store={store}>
          <RootContainer {...props} store={store} />
        </Provider>)
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
