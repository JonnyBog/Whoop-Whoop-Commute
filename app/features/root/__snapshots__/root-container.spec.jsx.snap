// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Root Container should render correctly 1`] = `
<withRouter(Connect(Root))
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
